fre/pp/configure_script_yaml.py

```diff
@@ -1,125 +1,142 @@
-
-  dumper = metomi.rose.config.ConfigDumper()
-  outfile = os.path.join(cylc_dir, "rose-suite.conf")
-  dumper(rose_suite, outfile)
-  print("  " + outfile)
-
-  outfile = os.path.join(cylc_dir, "app", "regrid-xy", "rose-app.conf")
-  dumper(rose_regrid, outfile)
-  print("  " + outfile)
-
-  outfile = os.path.join(cylc_dir, "app", "remap-pp-components", "rose-app.conf")
-  dumper(rose_remap, outfile)
-  print("  " + outfile)
-                  
+    """
+    Using a valid pp.yaml, the rose-app and rose-suite
+    configuration files are created in the cylc-src
+    directory. The pp.yaml is also copied to the
+    cylc-src directory.
+    """
+    e = experiment
+    p = platform
+    t = target
+    yml = yamlfile
+
+    # initialize rose suite config
+    rose_suite = metomi.rose.config.ConfigNode()
+    rose_suite.set(keys=['template variables', 'SITE'],              value='"ppan"')
+    rose_suite.set(keys=['template variables', 'CLEAN_WORK'],        value='True')
+    rose_suite.set(keys=['template variables', 'PTMP_DIR'],          value='"/xtmp/$USER/ptmp"')
+    rose_suite.set(keys=['template variables', 'DO_STATICS'],        value='True')
+    rose_suite.set(keys=['template variables', 'DO_TIMEAVGS'],       value='True')
+    rose_suite.set(keys=['template variables', 'DO_ATMOS_PLEVEL_MASKING'], value='True')
+    # disagreeable; these should be optional
+    rose_suite.set(keys=['template variables', 'DO_ANALYSIS_ONLY'],  value='False')
+    rose_suite.set(keys=['template variables', 'DO_MDTF'],  value='False')
+    rose_suite.set(keys=['template variables', 'PP_DEFAULT_XYINTERP'],  value='0,0')
+
+    # initialize rose regrid config
+    rose_regrid = metomi.rose.config.ConfigNode()
+    rose_regrid.set(keys=['command', 'default'], value='regrid-xy')
+
+    # initialize rose remap config
+    rose_remap = metomi.rose.config.ConfigNode()
+    rose_remap.set(keys=['command', 'default'], value='remap-pp-components')
+
+    # set some rose suite vars
+    rose_suite.set(keys=['template variables', 'EXPERIMENT'], value=f'"{e}"')
+    rose_suite.set(keys=['template variables', 'PLATFORM'], value=f'"{p}"')
+    rose_suite.set(keys=['template variables', 'TARGET'], value=f'"{t}"')
+
+    with open(yml,"r") as f:
+        y=yaml.safe_load(f)
+        validate_yaml(y)
+
+    for key,value in y.items():
+        if key == "rose-suite":
+            for suiteconfiginfo,dicts in value.items():
+                for configkey,configvalue in dicts.items():
+                    if configvalue is not None:
+                        k=configkey.upper()
+                        if k in ("HISTORY_DIR", "PP_DIR", "ANALYSIS"):
+                            # replace generic variables in pp yaml
+                            stem=e.split("_",1)[0]
+                            exp=e.split("_",1)[1]
+                            replacevars=configvalue.replace("$(stem)",stem).replace("$(expname)",exp).replace("$(platform)",p).replace("$(target)",t)
```

Comment:
Yep, looks good. The first-draft expansion of Canopy "FRE properties".

Except we can't expand "stem" until we have a place to retrieve it from (properties.yaml?), and I suggest using "name" instead of "expname" as that is exactly as Bronx FRE Properties named them (so we have a bridge to Bronx-like behavior..)